src/main/js/helpers/axis.js

```diff
@@ -249,6 +267,74 @@ const processTickValues = (ticks) => {
     }
     return ticks.map((t) => (utils.isDate(t) ? utils.parseDateTime(t) : t));
 };
+
+/**
+ * Gets the number of ticks on the axis based on the upper and lower limits
+ *
+ * @private
+ * @param {number} range - range of values (upperLimit - lowerLimits)
+ * @returns {number} returns number of ticks for that range, based on a predefined set
+ */
+const getTicksCountFromRange = (range) => {
+    if (range <= constants.AXISRANGE_1) return constants.DEFAULT_TICKSCOUNT - 4;
+    if (range <= constants.AXISRANGE_2) return constants.DEFAULT_TICKSCOUNT - 3;
+    if (range <= constants.AXISRANGE_3) return constants.DEFAULT_TICKSCOUNT - 2;
+    if (range <= constants.AXISRANGE_4) return constants.DEFAULT_TICKSCOUNT - 1;
+    return constants.DEFAULT_TICKSCOUNT;
+};
+
+/**
+ * Gets average number of ticks to be used based on the Y and Y2 axes
+ * * result from getTicksCountFromRange for Y and Y2 axes
+ *
+ * @private
+ * @param {number} rangeY - Y axis range (upperLimit - lowerLimit)
+ * @param {number} rangeY2 - Y2 axis range (upperLimit - lowerLimit)
+ * @returns {number} returns number of ticks to be rendered between the upper limits & lower limits with
+ */
+const getAverageTicksCount = (rangeY, rangeY2) => {
+    const yTicksCount = getTicksCountFromRange(rangeY);
+    const y2TicksCount = getTicksCountFromRange(rangeY2);
+
+    return Math.round((yTicksCount + y2TicksCount) / 2);
+};
+
+/**
+ * Generates an array of tick values for to be used as the
+ * * tick labels on the Y & Y2 axis.
+ *
+ * @private
+ * @param {number} lowerLimit - Lower limit of the Y or Y2 Axis
+ * @param {number} upperLimit - Upper limit of the Y or Y2 Axis
+ * @param {number} ticksCount - Number of ticks between the upper and lower limits
+ * @returns {(Array)} returns array of values to be used as tick labels
+ */
+const generateYAxesTickValues = (
+    lowerLimit,
+    upperLimit,
+    ticksCount = constants.DEFAULT_TICKSCOUNT
+) => {
+    ticksCount = Math.abs(ticksCount);
+    const tickValues = [];
+    const [newLowerLimit, newUpperLimit] = d3
+        .scaleLinear()
+        .domain([lowerLimit, upperLimit])
+        .nice()
+        .domain();
```

Comment:
What does this statement do? Can we add a comment here?

Reply:
This statement gets rounded off values for the upper and lower limits. I added a comment explaining it: [fix](https://github.com/cerner/carbon-graphs/pull/216/commits/6b0fcd338507d07f0d6d34bde63436eb3fb6d8a8)